framework/src/reporters/ReporterData.C

```diff
@@ -52,6 +52,53 @@ ReporterData::getPostprocessorNames() const
   return output;
 }
 
+DenseVector<Real>
+ReporterData::getAllPostprocessorValues() const
+{
+  DenseVector<Real> all_values;
+
+  std::vector<Real> & output = all_values.get_values();
+
+  for (const auto & name_context_pair : _context_ptrs)
+  {
+    if (name_context_pair.first.isPostprocessor())
+      output.push_back(
+          getReporterValue<PostprocessorValue>(name_context_pair.first.getCombinedName()));
```

Comment:
This should capture PPs and scalar reporter values.
```suggestion
          getReporterValue<Real>(name_context_pair.first.getCombinedName()));
```

Reply:
The above check `if (name_context_pair.first.isPostprocessor())` will be only for postprocessor, right?